bitnami/kafka/templates/secrets.yaml

```diff
@@ -0,0 +1,136 @@
+{{- $bootstrapServers := list }}
+{{- if not .Values.controller.controllerOnly }}
+  {{- range $i, $e := until (int .Values.controller.replicaCount) }}
+    {{- $controller := printf "%s-controller-%s.%s-headless.%s.svc.%s" (include "common.names.fullname" $) (print $i) (include "common.names.fullname" $) $.Release.Namespace $.Values.clusterDomain }}
+    {{- $host = append $host $controller }}
+    {{- $bootstrapServers = append $bootstrapServers (printf "%s:%s" $controller .Values.service.ports.client) }}
+  {{- end }}
+{{- end }}
+{{- range $i, $e := until (int .Values.broker.replicaCount) }}
+  {{- $broker := printf "%s-broker-%s.%s-headless.%s.svc.%s" (include "common.names.fullname" $) (print $i) (include "common.names.fullname" $) $.Release.Namespace $.Values.clusterDomain }}
+  {{- $host = append $host $broker }}
+  {{- $bootstrapServers = append $bootstrapServers (printf "%s:%s" $broker .Values.service.ports.client) }}
+{{- end }}
+{{- range $i, $e := until (len .Values.sasl.client.users) }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "common.names.fullname" $ }}-svcbind-user-{{ $i }}
+  namespace: {{ $.Release.Namespace | quote }}
+  labels: {{- include "common.labels.standard" $ | nindent 4 }}
+    {{- if $.Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if $.Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+type: servicebinding.io/kafka
+data:
+  provider: {{ print "bitnami" | b64enc | quote }}
+  type: {{ print "kafka" | b64enc | quote }}
+  username: {{ index .Values.sasl.client.users $i | b64enc | quote }}
+  password: {{ index .Values.sasl.client.passwords $i | b64enc | quote }}
+  host: {{ join "," $host | b64enc | quote }}
+  port: {{ print $port | b64enc | quote }}
+  bootstrap-servers: {{ join "," $bootstrapServers | b64enc | quote }}
+{{- end }}
+{{- else }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "common.names.fullname" . }}-svcbind
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+type: servicebinding.io/kafka
+data:
+  provider: {{ print "bitnami" | b64enc | quote }}
+  type: {{ print "kafka" | b64enc | quote }}
+  host: {{ join "," $host | b64enc | quote }}
+  port: {{ print $port | b64enc | quote }}
+  bootstrap-servers: {{ join "," $bootstrapServers | b64enc | quote }}
+{{- end }}
+{{- end }}
+{{- end }}
+{{- if .Values.kraft.enabled }}
+apiVersion: v1
```

Comment:
There seems to be a `---` separation above here

Reply:
Nice catch! Indeed I missed a `---` in here, thank you for your feedback